int32 allied_type
---
int32 sharpness
int32 error